.rubocop.yml

```diff
@@ -60,6 +60,11 @@ Metrics/ClassLength:
     - lib/reek/context_builder.rb
     - lib/reek/cli/options.rb
 
+Style/SafeNavigation:
```

Comment:
Without this exception rubocop will complain:

```
Offenses:

lib/reek/ast/sexp_extensions/methods.rb:60:11: C: Style/NumericPredicate: Use ReferenceCollector.new(self).num_refs_to_self.positive? instead of ReferenceCollector.new(self).num_refs_to_self > 0.
          ReferenceCollector.new(self).num_refs_to_self > 0
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/reek/report/base_report.rb:57:9: C: Style/NumericPredicate: Use total_smell_count.positive? instead of total_smell_count > 0.
        total_smell_count > 0
        ^^^^^^^^^^^^^^^^^^^^^
```

Letting Rubocop auto-fix is will cause this line:

```Ruby
loc && loc.line
```

to become this line:

```Ruby
loc&.line
```

The problem is that now our reek quality check fails with:

```
Failures:

  1) Reek source code lib/reek/ast/node.rb has no smells
     Failure/Error: expect(pathname).not_to reek

       Expected no smells, but got:
         NilCheck: Reek::AST::Node#line performs a nil-check
     # ./spec/quality/reek_source_spec.rb:7:in `block (4 levels) in <top (required)>'

  2) Reek source code lib/reek/ast/sexp_extensions/module.rb has no smells
     Failure/Error: expect(pathname).not_to reek

       Expected no smells, but got:
         NilCheck: Reek::AST::SexpExtensions::CasgnNode#defines_module? performs a nil-check
     # ./spec/quality/reek_source_spec.rb:7:in `block (4 levels) in <top (required)>'

Finished in 2.42 seconds (files took 0.87515 seconds to load)
128 examples, 2 failures

Failed examples:

rspec ./spec/quality/reek_source_spec.rb[1:77:1] # Reek source code lib/reek/ast/node.rb has no smells
rspec ./spec/quality/reek_source_spec.rb[1:84:1] # Reek source code lib/reek/ast/sexp_extensions/module.rb has no smells
```

I'd rather not defer our Reek 5 release any longer because of that but rather fix this in a follow-up issue afterwards.

Reply:
See #1375

Reply:
Agreed :+1: